scripts/clean_obsolete_strings.py

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env python3
+
+'''
+This script is used to remove obsolete strings from all locales,
+and reformat files accordingly to the source content (en-US).
+'''
+
+from compare_locales.parser import getParser
+from compare_locales.serializer import serialize
+import os
+import subprocess
+import sys
+
+
+def extractFileList(repository_path):
+    '''
+        Extract the list of supported files. Store the relative path and ignore
+        specific folders for products we don't want to modify (e.g. SeaMonkey).
+    '''
+
+    supported_formats = [
+        '.dtd',
+        '.properties',
+        '.ftl',
+    ]
+
+    excluded_folders = (
+        'calendar',
+        'chat',
+        'editor',
+        'extensions',
+        'mail',
```

Comment:
Should we include Thunderbird, since they follow the release cycles, unlike SeaMonkey?

Reply:
Yeah, I think `mail` would be good. CC @jfx2006.

Reply:
@jfx2006
I still have Thunderbird excluded for now. For context, this script is used to remove obsolete strings from the repository, based on the content of `gecko-strings`.

Reply:
Calendar and chat are now non-optional parts of Thunderbird, so I see no reason to exclude them. Editor is only used by Seamonkey now.

Reply:
Thanks, removed both. Since we're only looking at known files that are available in quarantine (not removing extra files), `extensions` is safe to clean up.